### ISSUE TITLE
createFixtureLoader in buidler-waffle is buggy

### ISSUE THREAD

**Issue by PaulRBerg** [nomiclabs/buidler#760]
## Description

The `createFixtureLoader` binding method may have not been set up correctly:

https://github.com/nomiclabs/buidler/blob/61c48aad14ab09d0bbc91b2001ca970b798c5e2f/packages/buidler-waffle/src/index.ts#L28-L30

No matter what argument I pass to `overrideWallets`, the `getWallets` method gets executed in `fixtures.ts`:

https://github.com/nomiclabs/buidler/blob/61c48aad14ab09d0bbc91b2001ca970b798c5e2f/packages/buidler-waffle/src/fixtures.ts#L30

And it fails because the BuidlerEVM is not a Waffle MockProvider. The root of the issue seems to be that the `overrideWallets` parameter is always undefined:

https://github.com/nomiclabs/buidler/blob/61c48aad14ab09d0bbc91b2001ca970b798c5e2f/packages/buidler-waffle/src/fixtures.ts#L17

## Steps to Reproduce

1. Fork [paulrberg/solidity-template](https://github.com/paulrberg/solidity-template)
2. `git fetch --all`
3. `git checkout createFixtureLoaderBug`
4. `yarn install`
5. `yarn run test`

You can optionally modify `node_modules/@nomiclabs/buidler-waffle/dist/fixtures.js` and console log the `overrideWallets` variable to see it undefined.

## Environment

+ @nomiclabs/buidler@1.4.4
+ @nomiclabs/buidle-waffle@2.0.0
+ typescript@3.9.6

**Comment by PaulRBerg** [comment]
I first thought that #756 fixes this, but it doesn't. The bug here seems related to the behaviour of the JavaScript `.bind` functionality - perhaps it's not interpreted correctly by TypeScript?

**Comment by PaulRBerg** [comment]
Commit [9476378](https://github.com/PaulRBerg/buidler/commit/94763780b8b718259374f5c6051a0a788a2897d2) fixed this. The issue had to do with a bogus `.call` function:

```ts
createFixtureLoader: buidlerCreateFixtureLoader.bind(buidlerWaffleProvider),
```

Thad had to be changed to this:

```ts
createFixtureLoader: buidlerCreateFixtureLoader.bind(undefined, buidlerWaffleProvider),
```

Thanks @alcuadrado for your support and for including the fix in [buidler-waffle@2.1.0](https://github.com/nomiclabs/buidler/releases/tag/buidler-waffle-v2.1.0).
